src/GafferImage/RankFilter.cpp

```diff
@@ -237,15 +238,11 @@ void RankFilter::compute( Gaffer::ValuePlug *output, const Gaffer::Context *cont
 				IECore::Canceller::check( context->canceller() );
 
 				// Fill array with all nearby samples
-				V2i o;
-				vector<float>::iterator pixelsIt = pixels.begin();
-				for( o.y = -radius.y; o.y <= radius.y; ++o.y )
-				{
-					for( o.x = -radius.x; o.x <= radius.x; ++o.x )
-					{
-						*pixelsIt++ = sampler.sample( p.x + o.x, p.y + o.y );
-					}
-				}
+				V2i corner = p - radius;
+				sampler.visitPixels(
+					Imath::Box2i( p - radius, p + radius + Imath::V2i( 1 ) ),
+					[corner, supportSize, &pixels]( float v, int x, int y ){ pixels[ ( y - corner.y ) * supportSize + x - corner.x ] = v; }
```

Comment:
It feels like this would be easier if `visitPixels()` guaranteed a scanline visit order. And it feels like this might be fairly achievable without additional overhead?